sim/midas/src/main/cc/bridges/test_finisher.h

```diff
@@ -0,0 +1,32 @@
+
+#ifndef __TEST_FINISHER_H
+#define __TEST_FINISHER_H
+//See LICENSE for license details.
+
+#ifdef TESTFINISHERBRIDGEMODULE_struct_guard
+
+#include "bridge_driver.h"
+
+class test_finisher_t: public bridge_driver_t
+{
+    public:
+        test_finisher_t(simif_t* sim, 
+				TESTFINISHERBRIDGEMODULE_struct * mmio_addrs,
+        unsigned int num_messages,
+        unsigned int* isErr,
+        const char* const* msgs);
+				~test_finisher_t();
+        virtual void init() {};
+        virtual void tick();
+        virtual void finish() {};
+        virtual bool terminate() { return test_done; };
+        virtual int exit_code() { return test_done ? 1 : 0; };
```

Comment:
This needs to be driver by `isErr` in some way. You might as well set in tick as you've done with test_done.